plugins/serializers/prometheus/collection_test.go

```diff
@@ -425,3 +636,209 @@ func TestCollectionExpire(t *testing.T) {
 		})
 	}
 }
+
+func TestExportTimestamps(t *testing.T) {
+	tests := []struct {
+		name     string
+		now      time.Time
+		age      time.Duration
+		input    []Input
+		expected []*dto.MetricFamily
+	}{
+		{
+			name: "histogram bucket updates",
+			now:  time.Unix(0, 0),
+			age:  10 * time.Second,
+			input: []Input{
+				{
+					metric: testutil.MustMetric(
+						"prometheus",
+						map[string]string{},
+						map[string]interface{}{
+							"http_request_duration_seconds_sum":   10.0,
+							"http_request_duration_seconds_count": 2,
+						},
+						time.Unix(15, 0),
```

Comment:
Now is set to 0 seconds but the metric timestamps are in the future at 15 and 20 seconds. Shouldn't now be greater than or equal the latest timestamp? Maybe it doesn't matter since this isn't testing expiration but it seems odd.

Same thing happens in the other table test (line 754).

Reply:
Not sure how I missed this question but finally noticed it.

It doesn't matter for this test as the interaction for expiration is between `now` and `addtime` and these tests are confirming that a metrics timestamp is exported. I believe theoretically you could have a service report metrics with timestamps to telegraf and have those reported timestamps be in the future if telegraf and the source of metrics disagreed on the answer to "what time is it?"

Regardless, I updated the tests to have both `now` and `addtime` be moved to be timestamps that are **after** the metric timestamps. I made `now` and `addtime` distinct from the metric timestamps so it continues to be clear where the exported timestamp is coming from (the metric timestamp and not `now` or `addtime`).

See commit https://github.com/influxdata/telegraf/pull/9139/commits/b29e41a4a7159c516219b974b7078757624a3c0f

Reply:
@reimda Thanks for taking the time to review and think about the test data!